test/functional/rpc_getbalances.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the getbalances RPC.
+
+- Generate a few blocks
+- Check for certain fields in returned JSON
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (assert_equal, assert_greater_than,
+                                assert_is_hash_string)
+
+class GetBalancesTest (BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        # Move the chain
+        self.nodes[0].generatetoaddress(2, self.nodes[0].get_deterministic_priv_key().address)
+        # TEST: JSON outputs
+        balances = self.nodes[0].getbalances()
+        assert_greater_than(balances['mine']['immature'], 0)
+        assert_equal(balances['mine']['trusted'], 0)
+        # Keep current best block hash & height
+        prev_hash = self.nodes[0].getbestblockhash()
+        prev_height = self.nodes[0].getblock(prev_hash)['height']
+        # Move the chain again
+        self.nodes[0].generatetoaddress(5, self.nodes[0].get_deterministic_priv_key().address)
```

Comment:
IMO this 2nd round is unnecessary. I think you could factor out a `test_getbalances` function that would call `getbalances` and assert the result (and maybe just use `getblockchaininfo` to get chain tip). Then here you would do
```
test_getbalances(self.nodes[0], ...)
generatetoaddress(...)
test_getbalances(self.nodes[0], ...)
```

Reply:
Meanwhile, I've moved the tests into `wallet_walletbalance`. There I'm just calling a single function to run all those tests `test_lastblock_json_field`

Please, check.